src/lib/swapper/swappers/ThorchainSwapper/utils/getL1ToLongtailQuote.ts

```diff
@@ -0,0 +1,174 @@
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - Unsupported chainId ${buyAsset.chainId}.`,
+        code: TradeQuoteError.UnsupportedChain,
+        details: { buyAssetChainId: buyAsset.chainId },
+      }),
+    )
+  }
+
+  const chainAdapterManager = getChainAdapterManager()
+  const sellChainId = sellAsset.chainId
+  const buyChainId = buyAsset.chainId
+
+  const nativeSellAssetId = chainAdapterManager.get(sellChainId)?.getFeeAssetId()
+  const nativeSellAsset = nativeSellAssetId ? assetsById[nativeSellAssetId] : undefined
+
+  const nativeBuyAssetId = chainAdapterManager.get(buyChainId)?.getFeeAssetId()
+  const nativeBuyAsset = nativeBuyAssetId ? assetsById[nativeBuyAssetId] : undefined
+
+  if (!nativeBuyAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${buyChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { buyAssetChainId: buyChainId },
+      }),
+    )
+  }
+
+  if (!nativeSellAsset) {
+    return Err(
+      makeSwapErrorRight({
+        message: `[getThorTradeQuote] - No native buy asset found for ${sellChainId}.`,
+        code: TradeQuoteError.InternalError,
+        details: { sellAssetChainId: sellChainId },
+      }),
+    )
+  }
+
+  // TODO: use more than just UniswapV3, and also consider trianglar routes.
+  const l1Tol1QuoteInput: GetTradeQuoteInput = {
+    ...input,
+    buyAsset: nativeBuyAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit: sellAmountCryptoBaseUnit,
+  }
+
+  const maybeThorchainQuotes = await getL1quote(
+    l1Tol1QuoteInput,
+    streamingInterval,
+    TradeType.L1ToLongTail,
+  )
+
+  if (maybeThorchainQuotes.isErr()) return Err(maybeThorchainQuotes.unwrapErr())
+
+  const thorchainQuotes = maybeThorchainQuotes.unwrap()
+
+  let bestAggregator: AggregatorContract
+  let quotedAmountOut: bigint
+
+  const promises = await Promise.allSettled(
+    thorchainQuotes.map(async quote => {
+      const onlyStep = quote.steps[0]
+
+      const maybeBestAggregator = await getBestAggregator(
+        nativeBuyAsset,
+        getWrappedToken(nativeBuyAsset),
+        getTokenFromAsset(buyAsset),
+        onlyStep.buyAmountAfterFeesCryptoBaseUnit,
+      )
+
+      if (maybeBestAggregator.isErr()) return Err(maybeBestAggregator.unwrapErr())
+
+      const unwrappedResult = maybeBestAggregator.unwrap()
+
+      bestAggregator = unwrappedResult.bestAggregator
+      quotedAmountOut = unwrappedResult.quotedAmountOut
+
+      const updatedMemo = addAggregatorAndDestinationToMemo({
+        aggregator: bestAggregator,
+        finalAssetAddress: fromAssetId(buyAsset.assetId).assetReference as Address,
+        minAmountOut: convertDecimalPercentageToBasisPoints(
+          slippageTolerancePercentageDecimal ??
+            getDefaultSlippageDecimalPercentageForSwapper(SwapperName.Thorchain),
+        ).toString(),
+        quotedMemo: quote.memo,
+      })
```

Comment:
As discussed, this seems off: https://dev.thorchain.org/aggregators/memos.html

The docs don't do a very good job at explaining what this does and the syntax, but the name is pretty self-explanatory - that's the minimum amount out, to be passed:
- from the router to the aggregator contract ([docs](https://dev.thorchain.org/aggregators/evm-implementation.html#swapout))
- from the aggregator contract to the AMM: https://etherscan.io/address/0xe4ddca21881bac219af7f217703db0475d2a9f02#code#F5#L1 (see `swapRouter.exactInputSingle(IUniswapRouterV3.ExactInputSingleParams(params));` there
- then, the AMM handles minAmountOut the usual way, see for UNI-V3: https://github.com/Uniswap/v3-periphery/blob/697c2474757ea89fec12a4e6db16a574fe259610/contracts/SwapRouter.sol#L128

In other words, `minAmountOut` shouldn't be a slippage, but a minimum expected amount out for the final asset i.e expected final asset amount out - (expected final asset amount out * slippage bps)